calico_node/filesystem/startup.py

```diff
@@ -259,9 +260,10 @@ def main():
     # Write a startup environment file containing the IP address that may have
     # just been detected.
     # This is required because the confd templates expect to be able to fill in
-    #  the IP by fetching it from the environment.
+    # some templates by fetching them from the environment.
     with open('startup.env', 'w') as f:
-        f.write("IP=" + ip)
+        f.write("export IP=%s\n" % ip)
```

Comment:
Is podIP ever different to nodeIP for a host networked pod?  If so when?  We may well end up depending on this for etcd anyway; so I'd like to understand any concrete problems with it.

calico-node detection may be useful, but it's also horrible for debugability: anything where we're having to guess what calico-node is doing inside is unpleasant, as you've found above.  Having parameters be auto-filled deep in the bowels of a container is just dubious design: they should be pinned down early - calico-node should depend as little as possible on the host it's running on.